ui-v2/app/services/repository/metrics.js

```diff
@@ -21,10 +21,19 @@ export default RepositoryService.extend({
     opts.metrics_proxy_enabled = uiCfg.metrics_proxy_enabled;
     // Inject the base app URL
     const provider = uiCfg.metrics_provider || 'prometheus';
-    this.provider = window.consul.getMetricsProvider(provider, opts);
+
+    try {
+      this.provider = window.consul.getMetricsProvider(provider, opts);
+    } catch(e) {
+      console.log(`Error initializing metrics provider '${provider}'`, e);
+    }
   },
 
   findServiceSummary: function(protocol, slug, dc, nspace, configuration = {}) {
+    if (!this.provider) {
+      // Error initializing the provider just return an error
+      return Promise.reject('metrics provider not intialized');
+    }
```

Comment:
Wonder if we should reject/throw error objects here? Also tiny typo in all of these strings (see further down also) you might want to fix up.

Reply:
Huh I'm confused because I change this this morning and made it store the error rather than console.log and then reject with the actual error here... but that code has gone locally. I've not even rebased since then I don't think but I must have messed something up. Will put it back!

Reply:
Undo in my editor brought it back... 🤷

Reply:
:magic: 😂